lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,535 @@
+
+        display.vvvv(u"_wrap_command: '{0}'".format(to_text(cmd)), host=self.host)
+        return cmd
+
+    def _post_process(self, stdout):
+        ''' extract command status and strip unwanted lines '''
+
+        if self.is_windows:
+            success = stdout.rfind('True')
+            fail = stdout.rfind('False')
+
+            if success > fail:
+                returncode = 0
+                stdout = stdout[:success]
+            elif fail > success:
+                try:
+                    # test using: ansible -m raw -a 'cmd /c exit 99'
+                    returncode = int(stdout[fail:].split()[1])
+                except (IndexError, ValueError):
+                    returncode = -1
+                stdout = stdout[:fail]
+            else:
+                returncode = -51
+
+            return (returncode, stdout)
+        else:
+            # Get command return code
+            returncode = int(stdout.splitlines()[-2])
+
+            # Throw away ending lines
+            for x in range(0, 3):
+                stdout = stdout[:stdout.rfind('\n')]
+
+            return (returncode, stdout)
+
+    def _filter_ansi(self, line):
+        ''' remove any ANSI terminal control codes '''
+        line = line.decode('utf-8')
+
+        if self.is_windows:
+            ansi_filter = re.compile(r'(\x9B|\x1B\[)[0-?]*[ -/]*[@-~]')
+            line = ansi_filter.sub('', line)
+
+            # Replace or strip sequence (at terminal width)
+            line = line.replace('\r\r\n', '\n')
+            if len(line) == 201:
+                line = line[:-1]
+
+        return line
+
+    def _flush_stderr(self, subprocess):
+        ''' read and return stderr with minimal blocking '''
+
+        poll_stderr = select.poll()
+        poll_stderr.register(subprocess.stderr, select.POLLIN)
+        stderr = ''
+
+        while subprocess.poll() is None:
+            if poll_stderr.poll(1):
+                line = subprocess.stderr.readline()
+                display.vvvv(u"stderr line: {0}".format(to_text(line)), host=self.host)
+                stderr = stderr + line
+            else:
+                break
+
+        return stderr
+
+    def _get_url(self, client_method, bucket_name, out_path, http_method):
+        ''' Generate URL for get_object / put_object '''
+        client = boto3.client('s3')
+        return client.generate_presigned_url(client_method, Params={'Bucket': bucket_name, 'Key': out_path}, ExpiresIn=3600, HttpMethod=http_method)
+
+    @_ssm_retry
+    def _file_transport_command(self, in_path, out_path, ssm_action):
+        ''' transfer a file from using an intermediate S3 bucket '''
+
+        s3_path = out_path.replace('\\', '/')
+        bucket_url = 's3://%s/%s' % (self.get_option('bucket_name'), s3_path)
+
+        if self.is_windows:
+            put_command = "curl -Method PUT -InFile '%s' -Uri '%s'" % (in_path, self._get_url('put_object', self.get_option('bucket_name'), s3_path, 'PUT'))
```

Comment:
You should not use `curl` as the alias for `Invoke-WebRequest`. Use the full cmdlet name to avoid confusing people whether it's actually using curl or a PowerShell cmdlet.

Reply:
Used windows specific commands in the commit https://github.com/ansible/ansible/commit/8a5f982ce77b02d747c2f1954c3e413a449211f8